### PR TITLE
Solo 111 Fix string var length block

### PR DESCRIPTION
**Pull request by MatzElectronics**
Addresses #111 

**Summary:**
 - Adds fixes from google (added as overrides until we update our core)
 - Some code and variable name cleanup/improvements

**This affects:**
 - wx_scan
 - string_var_length
 - scan_string
 - serial_receive_multiple
 - constant_define

**To Test:**
 - create a new project 
 - make 4 new variables with different names
 - create 2 or 3 named constants with the define constant block
 - drag each of the blocks out and edit them.  Add fields using their mutators.
 - rearrange the blocks in the mutator and make sure the values follow.
 - save and re-open, verify persistence
 - check code view/compile.  Some blocks require inits.
   - FWIW, I'm not worried about code generation - generators were not edited at all, and field names were not changed.
 - Additionally, spot check other blocks with mutators.  The fix from google changes how they render under the hood, so worth a quick check.

**Still investigating:**
Older projects with these blocks being opened back up.  Are the blocks reproduced faithfully?  The XML in those older files looks a bit fishy and I'm suspecting an off-by-1 error.  This would cause them to render as their defaults and not show what the user set them as.  If we see this I can implement a fix, but I'd rather not if we don't have to.

### CODE REVIEW

**Comment by MatzElectronics** [comment]
This one is ready to go.

File: src/blockly/generators/propc/base.js [hunk 1]
```diff
@@ -814,7 +813,7 @@ Blockly.Blocks.string_var_length = {
         this.appendDummyInput()
                 .appendField('String variable set size of');
         this.optionList_ = ['var'];
-        this.v_list = ['MYVALUE', 'MYVALUE'];
+        this.userDefinedConstantsList_ = ['MYVALUE'];
```
[the remark this answers is not in the document]
Comment (MatzElectronics) [reply src/blockly/generators/propc/base.js]: It's a key-value pair within the block definition.  We don't have a style guide or naming convention, so I'm always unsure if the key should contain an underscore or not.  It's private to the block object - it's not used outside of the block.

I'd find a style guide helpful...I'll add a new issue.

File: src/blockly/generators/propc.js [hunk 2]
```diff
@@ -732,4 +732,133 @@ function uniq_fast(a) {
     } catch (err) {
         return tmpOut;
     }
-}
\ No newline at end of file
+}
+
+// TODO: Remove the following overrides after updating to a blockly core with these patches (targeted for 2019Q4).
```
[the remark this answers is not in the document]
Comment (MatzElectronics) [reply src/blockly/generators/propc.js]: Done.

File: src/blockly/generators/propc/base.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pjewald) [reply src/blockly/generators/propc/base.js]: There is the [Google Javascript style guide](https://google.github.io/styleguide/javascriptguide.xml) that the Blockly team uses. The trailing underscore is a pattern that I know the Blockly team uses to denote a variable of function that is not a part of the public API.

**Comment by pjewald** [comment]
Running the test procedure after the PR; Do constant declaration blocks only support numeric values? They do not appear to support strings, which I find very odd.

**Comment by pjewald** [comment]
"drag each of the blocks out and edit them. Add fields using their mutators."
None of the blocks have mutators. Are there other blocks involved?

Screen Shot 2019-11-25 at 2 45 46 PM

**Comment by MatzElectronics** [comment]
Yes - they only support numeric values.  Handing strings in PropC can be
persnickety, and we opted to just handle INTS because it too easy for a
user to create something that wouldn't compile.

**Comment by MatzElectronics** [comment]
Yes - The string var length block (OPERATORS > STRINGS, first block) needs
to be mutated, and you can choose to set the length of a var to a named
constant.

**Pull request merged**